File: Codeforces/1550/C.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int t, n;
    cin >> t;
    auto bad = [](const vector<int> &arr) -> bool {
        int n = arr.size();
        for (int i = 0; i < n; ++i) {
            for (int j = i + 1; j < n; ++j) {
                for (int k = j + 1; k < n; ++k) {
                    if (arr[i] >= arr[j] and arr[j] >= arr[k]) {
                        return true;
                    }
                    if (arr[i] <= arr[j] and arr[j] <= arr[k]) {
                        return true;
                    }
                }
            }
        }
        return false;
    };
    while (t--) {
        cin >> n;
        vector<int> arr(n);
        for (auto &ele : arr) cin >> ele;
        int ans = 0;
        for (int i = 0; i < n; ++i) {
            vector<int> temp;
            for (int j = i; j < n; ++j) {
                temp.push_back(arr[j]);
                if (bad(temp)) break;
                ++ans;
            }
        }
        cout << ans << "\n";
    }
    return 0;
}

File: Codeforces/1550/C.kt
private fun readLn() = readLine() ?: ""
private fun readLns() = readLn().split(" ")
private fun readInt() = readLn().toInt()
private fun readInts() = readLns().map { it.toInt() }

private fun bad(arr: MutableList<Int>): Boolean {
    for (i in arr.indices) {
        for (j in i + 1 until arr.size) {
            for (k in j + 1 until arr.size) {
                if (arr[i] >= arr[j] && arr[j] >= arr[k]) {
                    return true
                }
                if (arr[i] <= arr[j] && arr[j] <= arr[k]) {
                    return true
                }
            }
        }
    }
    return false
}

fun main() {
    val t = readInt()
    repeat(t) {
        val n = readInt()
        val arr = readInts()
        var ans = 0
        for (i in arr.indices) {
            val temp = mutableListOf<Int>()
            for (j in i until arr.size) {
                temp.add(arr[j])
                if (bad(temp)) break
                ++ans
            }
        }
        println(ans)
    }
}